### PR TITLE
JAMES-1900 Decode issues with headers

### PR DESCRIPTION
**Pull request by chibenwa**
Problem found with : 

 - InMemory indexing
 - JMAP raw headers
 - JSieve
 - HasHeader matcher

Needs https://github.com/linagora/james-jsieve/pull/9

Provides unit tests for https://github.com/linagora/james-jsieve/pull/9

### CODE REVIEW

File: mailet/standard/src/main/java/org/apache/james/transport/matchers/HasHeader.java [hunk 1]
```diff
@@ -40,6 +43,12 @@
  */
 public class HasHeader extends GenericMatcher {
 
+    private static String sanitizeHeaderField(String headerName) {
```
Comment (quynhn) [inline mailet/standard/src/main/java/org/apache/james/transport/matchers/HasHeader.java]: A lot of time we see this. Could we have common utility somewhere?

File: server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/delivery/SieveIntegrationTest.java [hunk 2]
```diff
@@ -264,6 +265,25 @@ public void headerScriptShouldWorkIfHeaderIsAbsent() throws Exception {
         assertThat(mail.getAttribute(MailStore.DELIVERY_PATH_PREFIX + LOCAL_PART)).isEqualTo(expressMailboxNameWithSlash(NOT_SELECTED_MAILBOX.getName()));
     }
 
+    @Ignore("Waiting for JAMES-1900 JSieve part to be merged")
```
Comment (quynhn) [inline server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/delivery/SieveIntegrationTest.java]: To be notice

File: mailet/standard/src/test/resources/mime/headerFolded.mime [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+
+
+-------- Message transéré --------
+Sujet : 	Invitation: (Aucun objet) - ven. 20 janv. 2017 14:00 - 15:00
+(CET) (duprat@linagora.com)
+Date : 	Thu, 19 Jan 2017 19:18:23 +0000
+De : 	Antoine Duprat <duprat@gmail.com>
+RÃ©pondre Ã  : 	antduprat@gmail.com
+Pour : 	aduprat@linagora.com
```
Comment (mbaechler) [inline mailet/standard/src/test/resources/mime/headerFolded.mime]: who is this ?

File: mailet/standard/src/test/resources/mime/headerFolded.mime [hunk 4]
```diff
@@ -0,0 +1,154 @@
+Return-Path: <aduprat@linagora.com>
```
Comment (mbaechler) [inline mailet/standard/src/test/resources/mime/headerFolded.mime]: who is this ?

File: mailet/standard/src/main/java/org/apache/james/transport/matchers/HasHeader.java [hunk 5]
```diff
@@ -75,7 +84,7 @@ public boolean isMatching(MimeMessage mimeMessage) throws MessagingException {
             String[] headerArray = mimeMessage.getHeader(headerName);
```
Comment (mbaechler) [inline mailet/standard/src/main/java/org/apache/james/transport/matchers/HasHeader.java]: I definitely believe mimeMessage should handle header decoding

**Comment by chibenwa** [comment]
```
Is it mime4j messages?
```

Not always. It can also be javax mime messages (HasHeader...) However, we can share in mime4j-james-utils this piece of logic to avoid the duplication spotted by Quynh.

File: mailet/standard/src/main/java/org/apache/james/transport/matchers/HasHeader.java (hunk 5 above)
Comment (chibenwa) [inline mailet/standard/src/main/java/org/apache/james/transport/matchers/HasHeader.java]: It's a javax code out of our control...

File: server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/delivery/SieveIntegrationTest.java (hunk 2 above)
Comment (chibenwa) [inline server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/delivery/SieveIntegrationTest.java]: No, to be merged

File: mailet/standard/src/main/java/org/apache/james/transport/matchers/HasHeader.java (hunk 5 above)
Comment (mbaechler) [inline mailet/standard/src/main/java/org/apache/james/transport/matchers/HasHeader.java]: one more reason to get rid of javax.mail. Not for today, I agree.

**Comment by chibenwa** [comment]
test this please

**Pull request merged**